Mathlib/NumberTheory/Rayleigh.lean

```diff
@@ -0,0 +1,246 @@
+/-
+Copyright (c) 2023 Jason Yuen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jason Yuen
+-/
+import Mathlib.Data.Real.ConjugateExponents
+import Mathlib.Data.Real.Irrational
+
+/-!
+# Rayleigh's theorem on Beatty sequences
+
+This file proves Rayleigh's theorem on Beatty sequences. We start by proving `rayleigh_int`, which
+is a generalization of Rayleigh's theorem, and eventually prove `rayleigh_irr_pos`, which is
+Rayleigh's theorem.
+
+## Main definitions
+
+* `beattySequence`: In the Beatty sequence for real number `r`, the `k`th term is `⌊k * r⌋`.
+* `beattySequence'`: In this variant of the Beatty sequence for `r`, the `k`th term is
+  `⌈k * r⌉ - 1`.
+* `beattySet`: Define the Beatty set for `r ∈ ℝ` to be `B_r := {⌊k * r⌋ | k ∈ ℤ}`.
+* `beattySet'`: Define another Beatty set for `r ∈ ℝ` to be `B'_r := {⌈k * r⌉ - 1 | k ∈ ℤ}`.
+* `beattySetPos`: Define the positive Beatty set for `r ∈ ℝ` to be `B⁺_r := {⌊r⌋, ⌊2r⌋, ⌊3r⌋, ...}`.
+* `beattySetPos'`: Define another positive Beatty set for `r ∈ ℝ` to be
+  `B⁺'_r := {⌈r⌉-1, ⌈2r⌉-1, ⌈3r⌉-1, ...}`.
+
+## Main statements
+
+* `rayleigh_int`: Let `r` be a real number greater than 1.
+  Then every integer is in exactly one of `B_r` or `B'_(r / (r - 1))`.
+* `rayleigh_pos`: Let `r` be a real number greater than 1.
+  Then every positive integer is in exactly one of `B⁺_r` or `B⁺'_(r / (r - 1))`.
+* `rayleigh_irr_pos`: Let `r` be an irrational real number greater than 1.
+  Then every positive integer is in exactly one of `B⁺_r` or `B⁺_(r / (r - 1))`.
+* `rayleigh_int'`, `rayleigh_pos'`, `rayleigh_irr_pos'`: The corresponding theorem that uses
+  `Real.IsConjugateExponent r s` as a hypothesis.
+
+## References
+
+* [Wikipedia, *Beatty sequence*](https://en.wikipedia.org/wiki/Beatty_sequence)
+
+## Tags
+
+beatty, sequence, rayleigh, irrational, floor, positive
+-/
+
+/-- In the Beatty sequence for real number `r`, the `k`th term is `⌊k * r⌋`. -/
+noncomputable def beattySequence (r : ℝ) : ℤ → ℤ :=
+  fun k ↦ ⌊k * r⌋
+
+/-- In this variant of the Beatty sequence for `r`, the `k`th term is `⌈k * r⌉ - 1`. -/
+noncomputable def beattySequence' (r : ℝ) : ℤ → ℤ :=
+  fun k ↦ ⌈k * r⌉ - 1
+
+/-- Define the Beatty set for `r ∈ ℝ` to be `B_r := {⌊k * r⌋ | k ∈ ℤ}`. -/
+def beattySet (r : ℝ) : Set ℤ :=
+  { beattySequence r k | k }
+
+/-- Define another Beatty set for `r ∈ ℝ` to be `B'_r := {⌈k * r⌉ - 1 | k ∈ ℤ}`. -/
+def beattySet' (r : ℝ) : Set ℤ :=
+  { beattySequence' r k | k }
+
+/-- Define the positive Beatty set for `r ∈ ℝ` to be `B⁺_r := {⌊r⌋, ⌊2r⌋, ⌊3r⌋, ...}`. -/
+def beattySetPos (r : ℝ) : Set ℤ :=
+  { beattySequence r k | k > 0 }
+
+/-- Define another positive Beatty set for `r ∈ ℝ` to be `B⁺'_r := {⌈r⌉-1, ⌈2r⌉-1, ⌈3r⌉-1, ...}`. -/
+def beattySetPos' (r : ℝ) : Set ℤ :=
+  { beattySequence' r k | k > 0 }
+
+namespace Beatty
+
+variable {r : ℝ} (hr : r > 1) {j k : ℤ}
+
+theorem s_gt_1 : r.conjugateExponent > 1 := by
```

Comment:
This should be moved to `Mathlib.Data.Real.ConjugateExponents`

Reply:
Done. I added `Real.one_lt_conjugateExponent` to this file.